### PR TITLE
Plans: Improve flows for upgrading private sites to the Business plan

### LINKED ISSUE #37783: Plans: Correct issue with launching sites when a Business plan is purchased in signup.
When a Business plan is purchased in signup, the site is launched automatically.

This is confusing in the checklist and customer home because this step is already checked off. You also lose out on the interface showing you that your site is still private. This has been reassuring to folks in previous tests.

/cc @zeldadam because I know there is previous work happening here. Is there any easy way for us to avoid this in the short term?

### PR DESCRIPTION
**Pull request by jblz**
Since Business sites do not go Atomic until a specific action is performed (a plugin or theme is installed, for example), there's no need to gate the upgrade with a prompt or to set the site to public at upgrade time.  This change removes that behavior for Business plan upgrades, but keeps it for Online Store / eCommerce upgrades (because those are taken Atomic immediately).

We want to improve the UI for the post-upgrade pre-install plugin / theme screen, but that work is decoupled from this in #37868.

#### Changes proposed in this Pull Request

* Only show the Site Privacy Dialog for eCommerce plan Upgrades. No longer show it for Business Upgrades.

|Before|After|
|---|---|
|unlaunched-biz-plan-upgrade-before|unlaunched-biz-plan-upgrade-after|

* Only set eCommerce plans to public on checkout
* Add non breaking spaces between warning message & support text

|Before|After|
|---|---|
|Screen Shot 2019-11-22 at 10 47 20 AM|Screen Shot 2019-11-22 at 10 47 28 AM|

##### Test the Business plan upgrade

* Browse to `/plans` for an unlaunched free site
* Click the button to initiate a Business plan upgrade
  * You should not see a `Site Privacy` modal
  * You should be taken to checkout immediately
  * You should be able to complete checkout
  * Your site should remain unlaunched after checkout
* Browse to `/plugins`
  * You should be prevented from uploading or installing a plugin without launching your site
* Browse to `/theme`
  * You should be prevented from uploading or installing a theme without launching your site
* Launch your site
  * You should be able to install a theme or plugin

##### Test the eCommerce plan upgrade

* Browse to `/plans` for a separate unlaunched free site
* Click the button to initiate an eCommerce plan upgrade
  * You should see a `Site Privacy` modal
* Click `Let's do it`
  * You should be taken to checkout
  * You should be able to complete checkout
  * Your site should be launched after checkout

Fixes #37783

### CODE REVIEW

**Comment by jblz** [comment]
Rebased after landing #37641

File: client/blocks/site-launch-gate/index.tsx [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+						{ translate( 'Launch (but hide from Search Engines)' ) }
+					</Button>
+					<Button
+						primary
+						onClick={ e => {
+							e.preventDefault();
+							this.props.launchSite( siteId );
+						} }
+					>
+						{ translate( 'Launch Site' ) }
```
Comment (a8ci18n) [inline client/blocks/site-launch-gate/index.tsx]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **23** times:
`translate( 'Launch site' )` *ES Score: 11*

**Comment by adamziel** [comment]
This tests well and makes sense. The only note I have is I'm slightly concerned about sending more people to our HEs - the communication on `/plugins` could be much better, maybe even offer a one click solution. CC @jancavan @ianstewart

**Comment by jblz** [comment]
Thanks, @adamziel.

> the communication on /plugins could be much better, maybe even offer a one click solution.

Definitely.  I'm going to concentrate on that in #37868

**Comment by jancavan** [comment]
> the communication on /plugins could be much better, maybe even offer a one click solution.

Yes. This was one of the ideas explored in the Figma. We will also work on improving copy.

**Pull request merged**